src/Value/Location/Address/Street.php

```diff
@@ -38,6 +39,22 @@ public function __construct(StreetName $name, StreetNumber $number, ?StreetSuffi
         $this->suffix = $suffix ? (string) $suffix : null;
     }
 
+    /**
+     * @throws InvalidArgument
+     */
+    public static function fromSingleLine(string $streetAddress): self
+    {
+        if (\preg_match('/^(\D*[^\d\s]) *([\d]+)(.*)$/', $streetAddress, $results)) {
+            return new self(
+                StreetName::fromString($results[1]),
+                StreetNumber::fromString($results[2]),
+                !empty(\trim($results[3])) ? StreetSuffix::fromString($results[3]) : null
+            );
+        }
```

Comment:
This will give issues with street names that aren't "standard" such as `3e Haagstraat 135a` since `$result[1]` will be `3e` instead of `3e Haagstraat`. `StreetNumber::fromString($results[2])` will also become `Haagstraat` which isn't a valid number.

Reply:
Good find!

The explanation is wrong though. `$result[1]` will not contain `3e`, the regex simply won't parse at all because the street name can't start with a number (the `\D` part in the regex).

Reply:
Also make the solution for this compatible for street names that have multiple splits. Since we're preparing for France a lot of addresses like `Rue de X` will be stored in the system.